src/core/solver.py

```
@@ -108,67 +123,96 @@ class ScoringStrategy(ABC):
-                for c in cfg.score_weights
-            )
-            scores.append(round(total * config.SCALE_FACTOR))
-        return [("simple_total", scores, 100)]
+            vectors.append((col, scores, int(max(1, round(weight)))))
+        return vectors
 
 
 class ConstraintBuilder:
     """Stateful builder for constructing the CP-SAT partition model."""
 
     def __init__(self, participants: list[Participant], cfg: SolverConfig):
```

Ruff (0.15.12)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)